lib/tasks/job_runs.rake

```diff
@@ -2,8 +2,29 @@ namespace :job_runs do
   task run: :environment do
     warn 'Calling job runner. See rails log for output.'
 
-    loop do
-      JobRunner::Runner.new.run
+    keep_loop = true
+    puts "rake job_runs:run with PID #{Process.pid}" if keep_loop
+
+    `echo "#{Process.pid}" > #{Rails.root}/tmp/job_runs-0.pid`
+
+    def shut_down
+      `rm -rf #{Rails.root}/tmp/job_runs-0.pid`
+      keep_loop = false
+      puts "\nShutting down gracefully..." unless keep_loop
+    end
+
+    # Trap ^C
+    Signal.trap('INT') do
+      shut_down
+      exit
```

Comment:
`exit` causes the whole process to exit immediately instead of gracefully shutting down.